connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/KeyValueGroupedDataset.scala

```diff
@@ -588,16 +681,24 @@ private object KeyValueGroupedDatasetImpl {
     val dummyGroupingFunc = SparkUserDefinedFunction(
       function = UdfUtils.noOp[V, K](),
       inputEncoders = vEncoder :: Nil,
-      outputEncoder = kEncoder).apply(col("*"))
+      outputEncoder = kEncoder).withName(DUMMY_UDF_NAME).apply(col("*"))
     val session = df.sparkSession
     new KeyValueGroupedDatasetImpl(
       session,
       df.plan,
       kEncoder,
       vEncoder,
       vEncoder,
-      (Seq(dummyGroupingFunc) ++ groupingExprs).map(toExpr).asJava,
+      Seq(dummyGroupingFunc) ++ groupingExprs,
       None,
       () => df.select(groupingExprs: _*).as(kEncoder))
   }
+
+  val DUMMY_UDF_NAME = "__spark__connect__internal__dummy__"
```

Comment:
You can also check the instance. There is nothing mutable in the current UDF, so we might as well reuse it.

Reply:
Yep, it works.